src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -416,7 +427,9 @@ internal MethodSymbol GetLeastOverriddenMethod(NamedTypeSymbol accessingTypeOpt)
                 // See InternalsVisibleToAndStrongNameTests: IvtVirtualCall1, IvtVirtualCall2, IvtVirtual_ParamsAndDynamic.
                 MethodSymbol overridden = m.OverriddenMethod;
                 HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-                if ((object)overridden == null || !AccessCheck.IsSymbolAccessible(overridden, accessingType, ref useSiteDiagnostics))
+                if ((object)overridden == null ||
+                    !AccessCheck.IsSymbolAccessible(overridden, accessingType, ref useSiteDiagnostics) ||
+                    (requireSameReturnType && !this.ReturnType.Equals(overridden.ReturnType, TypeCompareKind.AllIgnoreOptions)))
```

Comment:
Does AllIgnoreOptions also ignore custom modifiers? We probably want to keep that, don't we?

Reply:
I don't think so. The language's overriding rules ignore custom modifiers. Their presence may cause the compiler to add them to the override.

---
In reply to: [437693763](https://github.com/dotnet/roslyn/pull/44025#discussion_r437693763) [](ancestors = 437693763)